src/core.js

```diff
@@ -821,6 +821,7 @@ $.extend( $.validator, {
 					group = this.groups[ element.name ];
 					if ( group ) {
 						$.each( this.groups, function( name, testgroup ) {
+							name = name.replace( /'/g, "\\'" );
 							if ( testgroup === group ) {
```

Comment:
Please move the escaping into the `if` more close to the place which requires the escaping